sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -32,12 +32,19 @@ import org.apache.spark.util.Utils
  * A logical plan representing a data source v2 table.
  *
  * @param table   The table that this relation represents.
+ * @param output the output attributes of this relation
+ * @param catalogIdentifier the string identifier for the catalog. None if no catalog is specified
+ * @param identifiers the identifiers for the v2 relation. For multipath dataframe, there could be
+ *                    more than one identifier or Nil if a V2 relation is instantiated using
+ *                    options
  * @param options The options for this table operation. It's used to create fresh [[ScanBuilder]]
  *                and [[WriteBuilder]].
  */
 case class DataSourceV2Relation(
     table: Table,
     output: Seq[AttributeReference],
+    catalogIdentifier: Option[String],
+    identifiers: Seq[Identifier],
```

Comment:
I don't think it is a good idea to have multiple identifiers here. DSv2 doesn't yet cover how file-based tables should work and I think we need a design document for them. Adding multiple identifiers here in support of something that has undefined behavior seems premature.

Design and behavior of path-based identifiers aside, a table should use one and only one identifier. When path-based tables are supported, I expect them to use a single `Identifier` with possibly more than one path embedded in it, like we do with the `paths` key.

Reply:
I see. If the specification is that there should be one and only one identifier, shall I just define it as `identifier: Identifier` instead of `identifier: Option[Identifier]`? The tricky part is still in `load(paths: String*)`, I might need to use some placeholder or `null` if we choose to not use Option. What do you guys think? cc @brkyvz

Reply:
Two things:
1. There should be only one identifier for a table because it is what identified the table
2. Let's use null for `load(paths: String*)` because path-based tables are not designed or supported in v2